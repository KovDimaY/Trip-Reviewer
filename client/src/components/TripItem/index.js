import React from 'react';
import { Link } from 'react-router-dom';

import { TRIPS } from '../../constants/routes';

const TripItem = item => (
  <Link to={`${TRIPS}/${item._id}`} className="book_item">
    <div className="book_header">
      <h2>
        {item.title}
      </h2>
    </div>
    <div className="book_items">
      <div className="book_author">
        {item.author}
      </div>

      <div className="book_bubble">
        <strong>
Price
        </strong>
        {' '}
$
        {item.price}
      </div>

      <div className="book_bubble">
        <strong>
Duration
        </strong>
        {' '}
        {item.duration}
        {' '}
days
      </div>

      <div className="book_bubble rating">
        <strong>
Rating
        </strong>
        {' '}
        {item.rating}
        {' '}
/ 5
      </div>

    </div>
  </Link>
);

export default TripItem;
