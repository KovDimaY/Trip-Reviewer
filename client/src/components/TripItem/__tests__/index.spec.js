import React from 'react';
import { create } from 'react-test-renderer';

import TripItem from '..';

jest.mock('react-router-dom', () => ({ Link: 'Link ' }));

const mockComponent = props => (
  <TripItem {...props} />
);

describe('<TripItem />', () => {
  it('should render component', () => {
    const props = {
      _id: '_id',
      author: 'author',
      price: 'price',
      duration: 'duration',
      rating: 'rating',
    };
    const tree = create(mockComponent(props)).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
