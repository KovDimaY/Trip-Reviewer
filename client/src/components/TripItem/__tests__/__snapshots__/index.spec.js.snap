// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TripItem /> should render component 1`] = `
<Link 
  className="book_item"
  to="/trips/_id"
>
  <div
    className="book_header"
  >
    <h2 />
  </div>
  <div
    className="book_items"
  >
    <div
      className="book_author"
    >
      author
    </div>
    <div
      className="book_bubble"
    >
      <strong>
        Price
      </strong>
       $ 
      price
    </div>
    <div
      className="book_bubble"
    >
      <strong>
        Duration
      </strong>
        
      duration
       days
    </div>
    <div
      className="book_bubble rating"
    >
      <strong>
        Rating
      </strong>
        
      rating
       / 5
    </div>
  </div>
</Link >
`;
